src/pineko/check.py

```diff
@@ -92,31 +93,100 @@ def is_fonll_b(fns, lumi):
     return False
 
 
-def contains_fact(grid):
-    """Check whether factorization scale-variations are available in the pineappl grid.
+def contains_fact(grid, max_as, max_al):
+    """Check whether factorization scale-variations are available in the pineappl grid for the requested order.
 
     Parameters
     ----------
         grid: pineappl.grid.Grid
             Pineappl grid
+        max_as: int
+            max as order
+        max_al: int
+            max al order
+    Returns
+    -------
+        : bool
+        is fact scale-variation available for as
+        : bool
+        is fact scale-variation available for al
     """
-    order_list = [order.as_tuple() for order in grid.orders()]
+    order_array = np.array([order.as_tuple() for order in grid.orders()])
+    order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
+    order_list = order_array[order_mask]
+    as_orders = []
+    al_orders = []
     for order in order_list:
-        if order[-1] != 0:
-            return
-    raise ValueError("Factorization scale variations are not available for this grid")
-
-
-def contains_ren(grid):
+        as_orders.append(order[0])
+        al_orders.append(order[1])
+    min_as = min(as_orders)
```

Comment:
Let's use the tools we already have - to find out `min_as` you can just call `create_mask` again

Reply:
Yes but `order_mask` is just returning a list full of `True` or `False` according to which order you need to take in the same order of the 'grid.orders()' list. So I believe that of course it is possible to get the `min_as` and the `min_al` values using `order_mask` but it would be less trivial than just using a loop and then the `min` func.

Reply:
as you prefer ...